hopsworks-common/src/main/java/io/hops/hopsworks/common/kafka/KafkaController.java

```diff
@@ -313,88 +319,126 @@ public SharedTopicsDTO shareTopicWithProject(Project project, String topicName,
     return dto;
   }
   
+  public Optional<TopicAcls> getTopicAclsByDto(String topicName, AclDTO dto) throws UserException{
+    Users user = Optional.ofNullable(userFacade.findByEmail(dto.getUserEmail()))
+      .orElseThrow(() ->
+        new UserException(RESTCodes.UserErrorCode.USER_WAS_NOT_FOUND, Level.FINE, "user: " + dto.getUserEmail()));
+  
+    String principalName = KafkaConst.buildPrincipalName(dto.getProjectName(), user.getUsername());
+    
+    return topicAclsFacade.getTopicAcls(topicName, dto, principalName);
+  }
+  
   private void addFullPermissionAclsToTopic(Integer aclProjectId, String topicName, Integer projectId)
-    throws ProjectException, KafkaException, UserException {
+    throws ProjectException {
     Project p = Optional.ofNullable(projectFacade.find(aclProjectId)).orElseThrow(() ->
       new ProjectException(RESTCodes.ProjectErrorCode.PROJECT_NOT_FOUND, Level.FINE,
       "Could not find project: " + aclProjectId));
     
-    addFullPermissionAclsToTopic(p.getName(), topicName, projectId);
+    p.getProjectTeamCollection()
+      .stream()
+      .map(member -> member.getUser().getEmail())
+      .map(email -> new AclDTO(p.getName(), email, "allow",
+        Settings.KAFKA_ACL_WILDCARD, Settings.KAFKA_ACL_WILDCARD, Settings.KAFKA_ACL_WILDCARD))
+      .forEach(acl -> addAclsToTopicSafe(topicName, projectId, acl));
```

Comment:
@kai-chi is this to not fail if an ACLs fails to be added for a member?

Reply:
this is an ugly Java workaround to use lambdas. methods that throw a checked exception can't be used in lambda expressions. it's wrapped in an unchecked exception though, so it's still gonna work the same. maybe the name is misleading?

Reply:
It's mostly that the ThrowableMapper will return a more generic error than the one it could return, or so I suppose. We can resort to plain old Java for this case?

Reply:
ok, will get rid of this method.
FYI, this is how it would work:
`try {
  return addAclsToTopic(topicName, projectId, dto);
} catch (ProjectException | KafkaException | UserException ex) {
  throw new RuntimeException(ex);
}`